Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM golang:alpine as builder
+# Set necessary environmet variables needed for our image
+ENV GO111MODULE=on \
+    CGO_ENABLED=0 \
+    GOOS=linux \
+    GOARCH=amd64
+
+# Move to working directory /build
+WORKDIR /build
+
+# Copy and download dependency using go mod
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the code into the container
+COPY . .
```

Comment:
Why not just copy entire code and then run `go mod download` instead of doing it twice?